packages/admin/admin-date-time/src/TimeRangePicker.tsx

```diff
@@ -0,0 +1,160 @@
+import { ComponentsOverrides, FormControl, Theme, Typography } from "@mui/material";
+import { createStyles, WithStyles, withStyles } from "@mui/styles";
+import clsx from "clsx";
+import * as React from "react";
+import { FormatDateOptions, FormattedMessage } from "react-intl";
+
+import { TimePicker, TimePickerProps } from "./TimePicker";
+
+export type TimeRange = {
+    start: string;
+    end: string;
+};
+
+export type TimeRangePickerIndividualPickerProps = Omit<TimePickerProps, "onChange" | "onBlur" | "value" | "className" | "inputRef">;
+
+export interface TimeRangePickerComponentsProps {
+    startPicker?: TimeRangePickerIndividualPickerProps;
+    endPicker?: TimeRangePickerIndividualPickerProps;
+}
+
+export interface TimeRangePickerProps {
+    onChange?: (timeRange?: TimeRange) => void;
+    value?: TimeRange;
+    formatOptions?: FormatDateOptions;
+    minuteStep?: number;
+    min?: string;
+    max?: string;
+    clearable?: boolean;
+    separatorText?: React.ReactNode;
+    componentsProps?: TimeRangePickerComponentsProps;
+}
+
+type IndividualTimeValue = string | undefined;
+
+function TimeRangePicker({
+    classes,
+    onChange,
+    value,
+    separatorText = <FormattedMessage id="cometAdmin.dateTime.fromToSeparatorText" defaultMessage="to" />,
+    componentsProps = {},
+    ...propsForBothTimePickers
+}: TimeRangePickerProps & WithStyles<typeof styles>) {
+    const [startTime, setStartTime] = React.useState<IndividualTimeValue>(value?.start);
+    const [endTime, setEndTime] = React.useState<IndividualTimeValue>(value?.end);
+
+    const [startPickerIsOpen, setStartPickerIsOpen] = React.useState<boolean>(false);
+    const [endPickerIsOpen, setEndPickerIsOpen] = React.useState<boolean>(false);
+
+    const startPickerRef = React.useRef<HTMLElement>(null);
+    const endPickerRef = React.useRef<HTMLElement>(null);
+
+    React.useEffect(() => {
+        if (startTime && endTime) {
+            onChange?.({ start: startTime, end: endTime });
+        } else if (startTime === undefined && endTime === undefined) {
+            onChange?.(undefined);
+        }
+    }, [startTime, endTime, onChange]);
+
+    React.useEffect(() => {
+        if (!startPickerIsOpen && !endPickerIsOpen) {
+            if (startTime !== undefined && endTime === undefined) {
+                setEndTime(startTime);
+            }
+            if (startTime === undefined && endTime !== undefined) {
+                setStartTime(endTime);
+            }
+        }
+    }, [startPickerIsOpen, endPickerIsOpen, startTime, endTime]);
+
+    const onChangeTimeValue = (newTimeValue: IndividualTimeValue, type: "start" | "end") => {
```

Comment:
I'd create two separate change handlers instead of adding the `type` param and four ifs. The two handlers might be even shorter than this combined handler 😉